Adds Navigation Menu based on Radix

kristian-committed opened the pull request:
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Bug fix
[ x ] Enhancement
[ ] Documentation
[ ] Deployment
[ ] Process (e.g. improve continuous integration)
[ ] Other, please explain:

**Which issue (if any) does this pull request address?**

Fixes #240

<img width="560" alt="Screenshot 2022-03-15 at 17 01 54" src="https://user-images.githubusercontent.com/81633615/158437111-812095e1-dda7-4a54-ba41-6ff62949af8c.png">
<img width="527" alt="Screenshot 2022-03-15 at 17 32 40" src="https://user-images.githubusercontent.com/81633615/158437301-996ebad3-b8c6-4236-83ac-9689b86bc899.png">
<img width="817" alt="Screenshot 2022-03-15 at 14 29 56" src="https://user-images.githubusercontent.com/81633615/158437099-0b407c5d-8d0f-46fb-9873-a236ac6a54f5.png">
<img width="578" alt="Screenshot 2022-03-15 at 17 01 42" src="https://user-images.githubusercontent.com/81633615/158437105-af8f3e4c-0ea5-415c-ab0c-34c04644c912.png">

stuarthendren commented on src/components/NavigationMenu/NavigationMenu.tsx:
```diff
@@ -0,0 +1,430 @@
+  padding: 4,
+  listStyle: 'none',
+  fontSize: '$0',
+  borderRadius: '$3',
```
Unless we have a really good reason all the borderRadius should be set to `'$default'`.

The valid token values come from the `stitches.config.ts` file property:

```
    radii: {
      default: '2px',
      round: '50%',
      pill: '9999px',
    },
```

So $3 here is not valid - and probably doesn't do anything.
Where you have used 4 below, that probably works but again I think we should at least try it with $default first - remember we are not trying to emulate the radix style but to fit with our style.

This also applies to the other properties too. So, for example we should not use 4 in the above padding, as this is not one of our design tokens.

stuarthendren commented on src/components/NavigationMenu/NavigationMenu.tsx:
```diff
@@ -0,0 +1,430 @@
+  display: 'flex',
+  justifyContent: 'center',
+  width: '100%',
+  zIndex: 1,
```
Is this required?
If so, we might want to add a token in the stitches config under zIndex for it - e.g. $nav, or $menu. However, I don't expect it to be required.

stuarthendren commented on src/components/NavigationMenu/NavigationMenu.tsx:
```diff
@@ -0,0 +1,430 @@
+  padding: '$2 $3',
+  outline: 'none',
+  userSelect: 'none',
+  borderRadius: 4,
```
use token

stuarthendren commented on src/components/NavigationMenu/NavigationMenu.tsx:
```diff
@@ -0,0 +1,430 @@
+  outline: 'none',
+  userSelect: 'none',
+  borderRadius: 4,
+  '&:focus': { position: 'relative', boxShadow: `0 0 0 $0 $colors$primary` },
```
use a shadow token - or if this is much better add it to the shadow tokens - don't forget to check how it looks in light and dark.

stuarthendren commented on src/components/NavigationMenu/NavigationMenu.tsx:
```diff
@@ -0,0 +1,430 @@
+      {...props}
+      ref={forwardedRef}
+      css={{
+        borderRadius: 3,
```
use token

stuarthendren commented:
Can I have new screen shots, in light and dark - don't have time to run it up here.

kristian-committed commented:
<img width="706" alt="Screenshot 2022-03-17 at 10 37 11" src="https://user-images.githubusercontent.com/81633615/158791807-0271924c-c025-4921-ba53-7520b3b7151b.png">
<img width="715" alt="Screenshot 2022-03-17 at 10 37 28" src="https://user-images.githubusercontent.com/81633615/158791815-11c24e99-fbd8-4583-84aa-ea4d9c5fb476.png">
<img width="737" alt="Screenshot 2022-03-17 at 10 37 55" src="https://user-images.githubusercontent.com/81633615/158791819-0d659a25-02cd-4135-9a40-569ad0025f37.png">
<img width="726" alt="Screenshot 2022-03-17 at 10 38 10" src="https://user-images.githubusercontent.com/81633615/158791821-5f227245-fca7-4a63-bb81-d2e48ccc8147.png">

kristian-committed commented on src/components/NavigationMenu/NavigationMenu.tsx:
```diff
@@ -0,0 +1,430 @@
+    boxShadow: 'none',
+  },
+  filter:
+    'drop-shadow(0px 3px 2.5px rgba(0,0,0,0.2)) drop-shadow(0px 6px 5px rgba(0,0,0,0.14)) drop-shadow(0px 1px 9px rgba(0,0,0,0.12))',
```
Didn't seem to be able to but as this filter doesn't add much, I've since removed it

stuarthendren commented:
can you add a small gap between the buttons ($2 probably).
And I'm a bit confused why the secondary button in the lower rows is larger the than the others. It shouldn't be.
I'm confused as it doesn't happen in the top rows. So that might imply it's something to do with the icon though I first thought it would be the border.

kristian-committed commented:
> can you add a small gap between the buttons ($2 probably). And I'm a bit confused why the secondary button in the lower rows is larger the than the others. It shouldn't be. I'm confused as it doesn't happen in the top rows. So that might imply it's something to do with the icon though I first thought it would be the border.

Added the gap and fixed sizing issue, turns out it was still the border causing the issue, the no-icon buttons are sized 100% to match up besides the icon ones so it wasn't noticeable. border-box doesn't seem to work how I expected, so I've added a transparent border to the non-secondary styles so that size is consistent.
